dali/pipeline/executor/queue_metadata.h

```diff
@@ -19,17 +19,31 @@
 
 namespace dali {
 
-struct QueueIdxs {
+// Used to store Stage queue sizes
+struct StageQueues {
   int &operator[](OpType op_type) { return idxs[static_cast<size_t>(op_type)]; }
 
   const int &operator[](OpType op_type) const { return idxs[static_cast<size_t>(op_type)]; }
 
-  explicit QueueIdxs(int uniform_idx) : idxs{uniform_idx, uniform_idx, uniform_idx, uniform_idx} {}
+  StageQueues() = default;
+
+  explicit StageQueues(int uniform_idx)
+      : idxs{uniform_idx, uniform_idx, uniform_idx, uniform_idx} {}
+
+  StageQueues(int support, int cpu, int mixed, int gpu) {
+    operator[](OpType::SUPPORT) = support;
```

Comment:
Somehow I find the following more readable:
```suggestion
    (*this)[OpType::SUPPORT] = support;
```

Reply:
I asked around, and the consensus was that operator[](...) is more readable.